### PR TITLE
added login controller and auth filter

### PR DESCRIPTION
**Pull request by TimoxaV**

### CODE REVIEW

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import com.internet.shop.service.UserService;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User user = userService.findByLogin(login).orElseThrow(() ->
```
Comment (tarkaiev) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: Optional - when we check login we can use .equalsIgnoreCase (don't use with password)

File: src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -40,4 +40,11 @@ public boolean deleteById(Long userId) {
     public List<User> getAll() {
         return Storage.users;
     }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        return Storage.users.stream()
+                .filter(user -> user.getLogin().equals(login))
```
Comment (tarkaiev) [inline src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java:47]: comment about equalsignorecase should be here))

**Review by tarkaiev — APPROVED** [review]
well done!

File: src/main/java/com/internet/shop/controllers/LoginController.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String pwd = req.getParameter("pwd");
+        try {
+            User user = authService.login(login, pwd);
+            HttpSession session = req.getSession();
+            session.setAttribute("user_id", user.getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/LoginController.java]: let's make user_Id a constant

File: src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java (hunk 2 above)
Comment (dmytro-hryhoruk) [reply src/main/java/com/internet/shop/dao/impl/UserDaoImpl.java:47]: I would choose to use equals()) but that'll be good if you will find some articles about any of these approach. WDYT, @tarkaiev ?

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+                new AuthenticationException("Incorrect login!"));
+        if (user.getPassword().equals(password)) {
+            return user;
+        }
+        throw new AuthenticationException("Incorrect password");
+    }
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: let's make both checks in one if and throw exception only once with message: incorrect login or password

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+public class AuthenticationFilter implements Filter {
+    private static final String USER_ID = "user_id";
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]:
```suggestion
```

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        if (userId == null || userService.get(userId).getId() == null) {
+            resp.sendRedirect("/login");
+            return;
+        }
+        filterChain.doFilter(req, resp);
+    }
+
+    @Override
+    public void destroy() {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]:
```suggestion
```

File: src/main/java/com/internet/shop/controllers/LoginController.java (hunk 3 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/controllers/LoginController.java]: done

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java (hunk 4 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: done

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java (hunk 5 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]: done

File: src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java (hunk 6 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/web/filters/AuthenticationFilter.java]: done

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User user = userService.findByLogin(login).orElse(null);
+        if (user == null || !user.getPassword().equals(password)) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: let's use ifPresent() check

**Review by dmytro-hryhoruk — APPROVED** [review]

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java (hunk 7 above)
Comment (TimoxaV) [reply src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: done

**Pull request merged**